geoh5py/objects/surveys/electromagnetics/base.py

```diff
@@ -337,19 +337,18 @@ def edit_metadata(self, entries: dict[str, Any]):
 
         :param entries: Metadata key value pairs.
         """
-        metadata = self.metadata.copy()
+        em_metadata = self.metadata.copy().pop("EM Dataset", {})
```

Comment:
We would almost need to change the name of this method `edit_em_metadata` to reflect what it does (i.e. only modify the 'EM Dataset' field of the metadata.)

Reply:
I will duplicate the function to have a deprecation warning for 0.10 for this method.